intf/intf.go

```diff
@@ -154,3 +168,56 @@ func AddIP(intf string, addr *net.IPNet) error {
 	}
 	return nil
 }
+
+// AwaitARP waits for the IPv4 address addr to be resolved via ARP. It uses the supplied
+// context to determine whether it should continue awaiting the entry. If the ARP entry
+// is resolved, it is returned directly.
+func AwaitARP(ctx context.Context, addr net.IP) (net.HardwareAddr, error) {
+	neighs, err := accessor.ARPList()
+	if err != nil {
+		return nil, fmt.Errorf("cannot retrieve list of ARP entries, %v", err)
+	}
+
+	for _, n := range neighs {
+		if addr.Equal(n.IP) {
+			return n.MAC, nil
+		}
+	}
+
+	updates := make(chan ARPUpdate, 1)
+	result := make(chan ARPUpdate, 1)
+	done := make(chan struct{})
+
+	go func(updates chan ARPUpdate, done chan struct{}) {
+		for {
+			select {
+			case upd := <-updates:
+				if upd.Type == ARPAdd {
+					if addr.Equal(upd.Neigh.IP) {
+						result <- upd
```

Comment:
how do closures work for golang? is it fine that you're writing to a channel that you didn't pass a pointer in for? Why did you need to pass updates and done into the function, but not resukt?

Reply:
It is fine to be writing to a channel this way - we can do both: refer to the channel outside of our scope, or hand it in to make it within the scope. In this case these two things are equivalent. 

Updated to make sure that this is consistent in this code.